### PR TITLE
Fixes #7957: Merge server-roles from initial promises and system technique

### PR DESCRIPTION
**Pull request by peckpeck**
https://www.rudder-project.org/redmine/issues/7957

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -30,6 +30,21 @@ all: rudder-templates-cli.jar
...
+	java -jar rudder-templates-cli.jar --outext .cf --outdir initial-promises/node-server/server-roles/1.0 techniques/system/server-roles/1.0/password-check.st
+	java -jar rudder-templates-cli.jar --outext .cf --outdir initial-promises/node-server/server-roles/1.0 techniques/system/server-roles/1.0/postgres-check.st
+	cp techniques/system/server-roles/1.0/rudder-logrotate.st initial-promises/node-server/server-roles/logrotate.conf/rudder
+	java -jar rudder-templates-cli.jar --outext .conf --outdir initial-promises/node-server techniques/system/server-roles/1.0/rudder-server-roles.st
+	java -jar rudder-templates-cli.jar --outext .cf --outdir initial-promises/node-server/server-roles/1.0 techniques/system/server-roles/1.0/servers-by-role.st
+	java -jar rudder-templates-cli.jar --outext .cf --outdir initial-promises/node-server/server-roles/1.0 techniques/system/server-roles/1.0/service-check.st
+	java -jar rudder-templates-cli.jar --outext .cf --outdir initial-promises/node-server/server-roles/1.0 techniques/system/server-roles/1.0/technique-reload.st
```
Comment (jooooooon) [inline Makefile]: Hmmm. This is a lot of identical lines - why don't we replace it by a find techniques/system/server-roles/1.0/ -name "*.st" -not -name (exceptions) -exec .. .?

File: initial-promises/node-server/server-roles/1.0/component-check.cf [hunk 2]
```diff
@@ -84,14 +84,14 @@ bundle agent root_component_check
...
+    # Do this if this is the root_server or a relay server
     root_server|(policy_server.!role_rudder_relay_promises_only)::
       "any" usebundle => root_networks_check;
       "any" usebundle => root_password_check_dav;
-      "any" usebundle => generic_process_check_process("${service[apache][binary]}", "${service[apache][initscript]}", "${service[apache][name]}", "false", "${service[apache][check_on_relay_server]}");
+      "any" usebundle => generic_process_check_process("${service[apache][binary]}", "${service[apache][service]}", "${service[apache][name]}", "${service[apache][check_on_relay_server]}");
+      "any" usebundle => generic_process_check_bootstart("${service[apache][binary]}", "${service[apache][service]}", "${service[apache][name]}");
 
-    !(root_server|(policy_server.!role_rudder_relay_promises_only))::
+    !(root_server|policy_server)|role_rudder_relay_promises_only::
```
Comment (jooooooon) [inline initial-promises/node-server/server-roles/1.0/component-check.cf]: This doesn't seem to quite match up

File: initial-promises/node-server/server-roles/1.0/logrotate-check.cf [hunk 3]
```diff
@@ -77,15 +77,15 @@ bundle agent root_logrotate_check
...
-      "@@server-roles@@result_success@@server-roles@@server-roles-directive@@0@@Check logrotate configuration@@None@@${g.execRun}##${g.uuid}@#The logrotate configuration is correct";
+      "any" usebundle => rudder_common_report("server-roles", "result_success", "server-roles@@server-roles-directive@@0", "Check logrotate configuration", "None", "The logrotate configuration is correct");
 
     rudder_logrotate_conf_copied.!cannot_copy_rudder_logrotate_conf::
-      "@@server-roles@@result_repaired@@server-roles@@server-roles-directive@@0@@Check logrotate configuration@@None@@${g.execRun}##${g.uuid}@#The logrotate configuration has been updated";
+      "any" usebundle => rudder_common_report("server-roles", "result_repaired", "server-roles@@server-roles-directive@@0", "Check logrotate configuration", "None", "The logrotate configuration has been updated");
 
-    cannot_copy_rudder_logrotate_conf::
-      "@@server-roles@@result_error@@server-roles@@server-roles-directive@@0@@Check logrotate configuration@@None@@${g.execRun}##${g.uuid}@#The logrotate configuration could not be updated";
+    annot_copy_rudder_logrotate_conf::
```
Comment (jooooooon) [inline initial-promises/node-server/server-roles/1.0/logrotate-check.cf]: Missing letter here: **c**annot_copy_rudder_logrotate_conf

**Comment by peckpeck** [comment]
PR rebased

File: Makefile (hunk 1 above)
Comment (peckpeck) [inline Makefile]: we cannot with current rudder-templates-cli.jar version

File: initial-promises/node-server/server-roles/1.0/component-check.cf [hunk 4]
```diff
@@ -85,14 +81,14 @@ bundle agent root_component_check
...
+    # Do this if this is the root_server or a relay server
     root_server|(policy_server.!role_rudder_relay_promises_only)::
       "any" usebundle => root_networks_check;
       "any" usebundle => root_password_check_dav;
-      "any" usebundle => generic_process_check_process("${service[apache][binary]}", "${service[apache][initscript]}", "${service[apache][name]}", "false", "${service[apache][check_on_relay_server]}");
+      "any" usebundle => generic_process_check_process("${service[apache][binary]}", "${service[apache][service]}", "${service[apache][name]}");
+      "any" usebundle => generic_process_check_bootstart("${service[apache][binary]}", "${service[apache][service]}", "${service[apache][name]}");
 
-    !(root_server|(policy_server.!role_rudder_relay_promises_only))::
+    !(root_server|policy_server)|role_rudder_relay_promises_only::
```
Comment (jooooooon) [inline initial-promises/node-server/server-roles/1.0/component-check.cf]: This still doesn't match up with the "positive" version above

**Pull request merged**